sklearn/model_selection/tests/test_search.py

```diff
@@ -1097,6 +1099,70 @@ def compare_refit_methods_when_refit_with_acc(search_multi, search_acc, refit):
         assert getattr(search_multi, key) == getattr(search_acc, key)
 
 
+@pytest.mark.parametrize('search_cv',
+                         [RandomizedSearchCV(estimator=DecisionTreeClassifier(),
+                                             param_distributions={'max_depth': [5, 10]}),
+                          GridSearchCV(estimator=DecisionTreeClassifier(),
+                                       param_grid={'max_depth': [5, 10]})
+                          ])
+def test_search_cv_score_samples_error(search_cv):
+    X, y = make_blobs(n_samples=100, n_features=4, random_state=42)
+    search_cv.fit(X, y)
+
+    # Make sure to error out when underlying estimator does not implement
+    # the method `score_samples`
+    err_msg = ("'DecisionTreeClassifier' object has no attribute "
+               "'score_samples'")
+
+    with pytest.raises(AttributeError, match=err_msg):
+        search_cv.score_samples(X)
+
+
+@pytest.mark.parametrize('search_cv',
+                         [RandomizedSearchCV(estimator=LocalOutlierFactor(novelty=True),
+                                             param_distributions={
+                                                 'n_neighbors': [5, 10]},
+                                             scoring="precision"),
+                          GridSearchCV(estimator=LocalOutlierFactor(novelty=True),
+                                       param_grid={'n_neighbors': [5, 10]},
+                                       scoring="precision")
+                          ])
+def test_search_cv_score_samples_method(search_cv):
+    # Set parameters
+    rng = np.random.RandomState(42)
+    n_samples = 300
+    outliers_fraction = 0.15
+    n_outliers = int(outliers_fraction * n_samples)
+    n_inliers = n_samples - n_outliers
+
+    # Create dataset
+    X = make_blobs(n_samples=n_inliers, n_features=2, centers=[[0, 0], [0, 0]],
+                   cluster_std=0.5, random_state=0)[0]
+    # Add some noisy points
+    X = np.concatenate([X, rng.uniform(low=-6, high=6,
+                                       size=(n_outliers, 2))], axis=0)
+
+    # Define labels to be able to score the estimator with `search_cv`
+    y_true = np.array([1] * n_samples)
+    y_true[-n_outliers:] = [-1] * n_outliers
+
+    # Fit on data
+    search_cv.fit(X, y_true)
+
+    # Define estimator with the best params obtained in
+    # the *SearchCV
+    lof = LocalOutlierFactor(novelty=True,
+                             n_neighbors=search_cv.best_params_["n_neighbors"]
+                             )
+
```

Comment:
Defining `LocalOutlierFactor` here is not needed anymore.

Reply:
ah, i see. it's stored as the best estimator below